client/implementation/src/main/java/io/smallrye/graphql/client/impl/typesafe/reflection/TypeInfo.java

```diff
@@ -447,4 +466,21 @@ public boolean isAnnotated(Class<? extends Annotation> type) {
     public <T extends Annotation> T getAnnotation(Class<T> type) {
         return ((Class<?>) this.type).getAnnotation(type);
     }
+
+    public TypeInfo subtype(String typename) {
+        return subtypes()
+                .filter(c -> c.getGraphQlTypeName().equals(typename))
+                .findAny()
+                .orElseThrow(() -> SmallRyeGraphQLClientMessages.msg.cannotInstantiateDomainObject(typename, null));
+    }
+
+    public Stream<TypeInfo> subtypes() {
+        var jsonbTypeInfo = getAnnotation(JsonbTypeInfo.class);
```

Comment:
Could we support the `@JsonTypeInfo`  from Jackson too? In Quarkus, there's a general movement to move away from JSON-B/P in favor of Jackson. For most annotations, we support the JSON-B as well as Jackson ones.

Reply:
We talked about that before. I understand that Yasson is notorious for its slowness. But wouldn't it be wrong to blame that on JSON-B? That's just the API, and the implementation doesn't _have to be_ slow, does it?

Is there an open discussion about this somewhere?

Reply:
See https://github.com/quarkusio/quarkus/issues/42748 
But yeah this is probably not that important right now, the move to a general SPI is still to be thought through and AFAIK we don't have a compile-time Jackson dependency, so outside the `client-model-builder` module, without Jandex, we would have to do it reflectively.